CMakeLists.txt

```diff
@@ -184,14 +185,15 @@ if(BUILD_ERROR_ON_WARNING)
     target_compile_options(${PROJECT_NAME}_objlib PRIVATE -Werror)
 endif()
 
-target_compile_options(${PROJECT_NAME}_objlib PRIVATE $<$<CONFIG:RELEASE>:-s>)
+# target_compile_options(${PROJECT_NAME}_objlib PRIVATE $<$<CONFIG:RELEASE>:-s>)
+# try to make it optional. removed as it will break debuginfo generation.
```

Comment:
I can't find documentation on the `-s` option, but overall I agree that all of these compile flags should be moved outside of the project configuration, since these affect downstream packagers. Use `CMakePresets.json` instead. If specific compile flags are indeed needed, please document them and guard them by the compiler ID/family.

For now though, we can circumvent this issue by specifying `-DCMAKE_BUILD=RelWithDebInfo` downstream, and minimize the changes of this PR  a bit.